Automattic/woocommerce-payments
# Track New Order Creation

Fixes 504-gh-Automattic/woocommerce-payments-server

#### Changes proposed in this Pull Request

* Adds the `ActionScheduler` library to `composer.json` (please note, this library is already bundled into WooCommerce, so it should already be installed on any site which uses WC Payments, however added here to provide code completion etc. as we are using it to send jobs to the server.
* Adds logic to create an ActionScheduler Job when an order is created via WooCommerce Payments.
* Adds logic to send an API request to the server via ActionScheduler.

#### Testing instructions

* Checkout the branch.
* Create an order in your store.
* Verify that you can see an action of type  `wcpay_track_new_order` in the **Tools -> Scheduled Actions** section of `wp-admin`. (It should be under `pending` actions if it hasn't already been run by AS).
* Click `Run` to run the action. Check the logs for your site in **WooCommerce -> Status -> Logs**. Verify that a POST request was made to the Payments API to the `/tracking/new_order` endpoint. Note that you'll also need the corresponding server PR to verify that the serverside logic is working as expected (but this PR only covers the client side, so just sending the request is the expected behaviour here).
* Confirm the unit tests pass with `npm run:test`

-------------------
